### PR TITLE
KAFKA-17434: Do not test impossible scenarios in upgrade_test.py

### PR DESCRIPTION
**Pull request by cmccabe**
Because of KIP-902 (Upgrade Zookeeper version to 3.8.2), it is not possible to upgrade from a Kafka version earlier than 2.4 to a version later than 2.4. Therefore, we should not test these upgrade scenarios in upgrade_test.py. They do happen to work sometimes, but only in the trivial case where we don't create topics or make changes during the upgrade (which would reveal the ZK incompatibility). Instead, we should test only supported scenarios.

### CODE REVIEW

File: tests/kafkatest/tests/core/downgrade_test.py [hunk 1]
```diff
@@ -19,8 +19,7 @@
 
 from kafkatest.services.kafka import config_property
 from kafkatest.tests.end_to_end import EndToEndTest
-from kafkatest.version import LATEST_1_1, LATEST_2_0, LATEST_2_1, LATEST_2_2, LATEST_2_3, LATEST_2_4, LATEST_2_5, \
-    LATEST_2_6, LATEST_2_7, LATEST_2_8, LATEST_3_0, LATEST_3_1, LATEST_3_2, LATEST_3_3, LATEST_3_4, LATEST_3_5, \
+from kafkatest.version import LATEST_3_5, \
     LATEST_3_6, LATEST_3_7, LATEST_3_8, DEV_BRANCH, KafkaVersion
```
Comment (jsancio) [inline tests/kafkatest/tests/core/downgrade_test.py]: Might as well join these two lines now that there are less imports.

File: tests/kafkatest/tests/core/upgrade_test.py [hunk 2]
```diff
@@ -24,9 +24,7 @@
 from kafkatest.tests.produce_consume_validate import ProduceConsumeValidateTest
 from kafkatest.utils import is_int
 from kafkatest.utils.remote_account import java_version
-from kafkatest.version import LATEST_0_8_2, LATEST_0_9, LATEST_0_10, LATEST_0_10_0, LATEST_0_10_1, LATEST_0_10_2, \
-    LATEST_0_11_0, LATEST_1_0, LATEST_1_1, LATEST_2_0, LATEST_2_1, LATEST_2_2, LATEST_2_3, LATEST_2_4, LATEST_2_5, \
-    LATEST_2_6, LATEST_2_7, LATEST_2_8, LATEST_3_0, LATEST_3_1, LATEST_3_2, LATEST_3_3, LATEST_3_4, LATEST_3_5, \
+from kafkatest.version import LATEST_3_5, \
     LATEST_3_6, LATEST_3_7, LATEST_3_8, V_0_11_0_0, V_2_8_0, V_3_0_0, DEV_BRANCH, KafkaVersion
```
Comment (jsancio) [inline tests/kafkatest/tests/core/upgrade_test.py]: Might as well join these two lines now that there are less imports.

**Review by jsancio — COMMENTED** [review]
Interesting. This is an issue because the ZK cluster provisioned by the system tests is the latest version and not a version that supports the old ZK clients?

File: tests/kafkatest/tests/core/upgrade_test.py (hunk 2 above)
Comment (cmccabe) [reply tests/kafkatest/tests/core/upgrade_test.py]: ok

**Comment by cmccabe** [comment]
> Interesting. This is an issue because the ZK cluster provisioned by the system tests is the latest version and not a version that supports the old ZK clients?

Unfortunately, there is no single ZK version that could support both old and new clusters. A version of ZK that worked with Kafka 0.8 would not work with Kafka 3.9.

For versions in between those two extremes, it's messy... there are more details in KIP-902.

**Comment by jsancio** [comment]
@cmccabe KIP-902 has the following:
> Zookeeper 3.8.2 server supports clients no older than 3.5.x (i.e. Kafka version 2.4.0) and Zookeeper 3.8.2. clients support server versions no older than 3.5.x (i.e. Kafka version 2.4.0).

Did you mean to allow upgrades from 2.4.0 or greater to the dev branch? The 3.5.x version is referring to the ZK version not the Kafka version.

**Comment by cmccabe** [comment]
Yes, I think you're right. It should be AK 2.4 that's the cutoff. not AK 3.5. Will fix.

File: tests/kafkatest/tests/core/downgrade_test.py [hunk 3]
```diff
@@ -19,9 +19,9 @@
 
 from kafkatest.services.kafka import config_property
 from kafkatest.tests.end_to_end import EndToEndTest
-from kafkatest.version import LATEST_1_1, LATEST_2_0, LATEST_2_1, LATEST_2_2, LATEST_2_3, LATEST_2_4, LATEST_2_5, \
+from kafkatest.version import LATEST_2_4, LATEST_2_5, \
     LATEST_2_6, LATEST_2_7, LATEST_2_8, LATEST_3_0, LATEST_3_1, LATEST_3_2, LATEST_3_3, LATEST_3_4, LATEST_3_5, \
-    LATEST_3_6, LATEST_3_7, LATEST_3_8, DEV_BRANCH, KafkaVersion
+    LATEST_3_6, LATEST_3_7, LATEST_3_8, V_0_11_0_0, V_2_8_0, V_3_0_0, DEV_BRANCH, KafkaVersion
```
Comment (jsancio) [inline tests/kafkatest/tests/core/downgrade_test.py]: I don't think the symbols `V_0_11_0_0`, `V_2_8_0`, `V_3_0_0` are used in this file.

**Review by jsancio — COMMENTED** [review]
@cmccabe are you planning to run Confluent's system test job on this branch/pr?

**Comment by cmccabe** [comment]
> @cmccabe are you planning to run Confluent's system test job on this branch/pr?

I ran a few variants locally using ducker and it seems good.

**Review by jsancio — APPROVED** [review]
LGTM

**Pull request merged**